enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,327 @@
+
+More information about the different format can be found at the URLs below:
+
+- Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+- Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The Operator SDK will traverse the given packagemanifest directory and for each
+of the versions create a new bundle.
+
+The Operator SDK will make use of the API available from the Operator
+Registry's `opm` command. The command is pretty well segmented that we could
+easily reuse the `GenerateFunc` function to create the `bundle` as a
+directory. There is also support for building the bundle as an image which
+we could do using the `BuildBundleImage` function.
+
+#### Generating bundle to disk
+
+The `GenerateFunc` takes in the following arguments:
+
+- directory string - local directory where the manifests and metadata are
+  located
+- outputDir string - where manifests and metadata are copied
+- packageName string - the name of the package that bundle image belongs to
+- channels string - the list of channels that bundle image belongs to
+- channelDefault string - the default channel for the bundle image
+- overwrite bool - flag to enable overwriting annotations.yaml
+
+In order to use this function, we will obtain the required parameters from the
+listed *source*:
+
+| parameter | source |
+| --------- | ------ |
+| directory | package manifest directory |
+| outputDir | default `bundle` dir or flag `--bundle-dir` |
+| packageName | packageName from `*.package.yaml` |
+| channels | ??? |
+| channelDefault | defaultChannel from `*.package.yaml` or `stable` |
+| overwrite | `--overwrite` parameter |
+
+#### Creating bundle image
+
+If we choose to build bundle images, then we can leverage the `BuildBundleImage`
+function from `opm`. This takes in two arguments:
+
+- imageTag string - tag to use for your container image
+- imageBuilder string - the container image builder i.e. `docker`, `podman`, etc
+
+This would output a new bundle image.
+
+In order to use this function, we will obtain the required parameters from the
+listed *source*:
+
+| parameter | source |
+| --------- | ------ |
+| imageTag | version of packagemanifest |
+| imageBuilder | `docker` |
+
+#### Command Line Interface
+
+The new command line will look as follows:
+
+```
+operator-sdk migrate bundle <packagemanifestdir> [--build-image=] \
+    [--bundle-dir=] [--overwrite]
+```
+
+- `<packagemanifestdir>` is a positional argument that specifies the
+  packagemanifest directory.
+- `[--build-image=]` is an optional flag that indicates whether we want to build
+  the bundle as an image or just output to a directory, defaults to `bundle`
+  directory.
+- `[--bundle-dir=]` is an optional flag that indicates the directory to write the
+  bundle to, defaults to `bundle` directory.
+- `[--overwrite]` is an optional flag that indicates whether we should overwrite
```

Comment:
In what cases will it make sense to use the `--overwrite` flag?

Reply:
Since this command is only intended to be run once, lets simplify things and say the command will delete the existing directory and recreate it from scratch no matter what, i.e. always overwrite.

Reply:
@estroz I like that idea.